Userland/Libraries/LibJS/Bytecode/Pass/GenerateCFG.cpp

```diff
@@ -8,99 +8,157 @@
 
 namespace JS::Bytecode::Passes {
 
-void GenerateCFG::perform(PassPipelineExecutable& executable)
+struct UnwindFrame {
+    BasicBlock const* handler;
+    BasicBlock const* finalizer;
+};
+
+static HashTable<BasicBlock const*> seen_blocks;
+static Vector<UnwindFrame*> unwind_frames;
+
+static void generate_cfg_for_block(BasicBlock const& current_block, PassPipelineExecutable executable)
```

Comment:
The depth of a callstack _can_ get pretty large, so if possible, I'd like to avoid recursively processing the CFG.
but what is this commit actually fixing? please mention that in the commit description.

Reply:
I added a note to the end of the commit message, but I agree to your point

Reply:
If you can come up with a solution in which we do not need to recurse, while still having accurate state management, I'd be happy to learn about it

Note: To my knowledge do some compilers just give up on blocks in try-catch-finally statements, which would also be a valid option, but we would need to forward this information to the next passes somehow.